pkg/controller/scyllacluster/sync_jobs.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/internalapi"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/resourceapply"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (scc *Controller) syncJobs(
+	ctx context.Context,
+	sc *scyllav1.ScyllaCluster,
+	services map[string]*corev1.Service,
+	jobs map[string]*batchv1.Job,
+) ([]metav1.Condition, error) {
+	var progressingConditions []metav1.Condition
+	var errs []error
+
+	requiredJobs, err := MakeJobs(sc, services, scc.operatorImage)
+	if err != nil {
+		return progressingConditions, fmt.Errorf("can't make jobs: %w", err)
+	}
+
+	err = controllerhelpers.Prune(ctx, requiredJobs, jobs,
+		&controllerhelpers.PruneControlFuncs{
+			DeleteFunc: scc.kubeClient.BatchV1().Jobs(sc.Namespace).Delete,
+		},
+		scc.eventRecorder,
+	)
+	if err != nil {
+		return progressingConditions, fmt.Errorf("can't prune job(s): %w", err)
+	}
+
+	var progressingMessages []string
+
+	for _, svc := range services {
+		if svc.Labels[naming.ScyllaServiceTypeLabel] != string(naming.ScyllaServiceTypeMember) {
+			continue
+		}
+
+		_, ok := svc.Annotations[naming.CurrentTokenRingHashAnnotation]
+		if !ok {
+			progressingMessages = append(progressingMessages, fmt.Sprintf("Service %q is missing current token ring hash annotation", naming.ObjRef(svc)))
+		}
+
+		_, ok = svc.Annotations[naming.LastCleanedUpTokenRingHashAnnotation]
+		if !ok {
+			progressingMessages = append(progressingMessages, fmt.Sprintf("Service %q is missing latest token ring hash annotation", naming.ObjRef(svc)))
+		}
+	}
+
+	if apimeta.IsStatusConditionTrue(sc.Status.Conditions, statefulSetControllerProgressingCondition) {
+		progressingMessages = append(progressingMessages, fmt.Sprintf("Waiting for ScyllaCluster %q to rollout.", naming.ObjRef(sc)))
+	}
+
+	if len(progressingMessages) != 0 {
```

Comment:
note: being discussed on the proposal - https://github.com/scylladb/scylla-operator/pull/1300#discussion_r1276433392